Education_Pathways/frontend/src/components/ResultDisplay.jsx

```diff
@@ -27,30 +30,38 @@ class SearchResultDisplay extends Component{
     this.getData(this.state.input)
   }
 
+  handleSubmit2(event) {
+    event.preventDefault();
+    this.getData("hi")
+  }
+
   getData = (input) => {
-    axios.get(`https://assignment-1-starter-template.herokuapp.com/searchc?input=${input}`)
+    API.get(`/searchc?input=${input}`)
+    // axios.get(`https://assignment-1-starter-template.herokuapp.com/searchc?input=${input}`)
       .then(res => {
-        console.log(`it is ${res.status}`)
+        // console.log(`it is ${res.status}`)
         if (res.status === 200) {
           this.setState({results: []})
-          
+          console.log(res.data.length)
           if (res.data.length > 0) {
             let len = res.data.length
             let result_temp = []
             result_temp.push(<Label></Label>)
             for (let i = 0; i < len; i++) {
-                result_temp.push(<Result course_code={res.data[i].code} course_name={res.data[i].name}></Result>)
+                result_temp.push(<Result key={res.data[i]._id} course_code={res.data[i].code} course_name={res.data[i].name}></Result>)
             }
             this.setState({results: result_temp})
-          } else if (res.data.length === 0) {
-            alert("Course not found")
-          }else {
-            let result_temp = []
-            result_temp.push(<Label></Label>)
-            result_temp.push(<Result course_code={res.data.course.code} course_name={res.data.course.name}></Result>)
-            this.setState({results: result_temp})
-          }
-
+          } 
+          else
+            if (res.data.length === 0) {
+              alert("Course not found")
+            }
+            else {
+              let result_temp = []
+              result_temp.push(<Label></Label>)
+              result_temp.push(<Result key={res.data.course._id} course_code={res.data.course.code} course_name={res.data.course.name}></Result>)
+              this.setState({results: result_temp})
+            }
         } else if (res.status === 400) {
           alert("System Error. Please refresh")
         }
```

Comment:
We should have an else statement in case of the occurrence of any other error (such as 500, which can happen due to a bug in our code). I suggest `else {
 alert("An unexpected error has occured")
}`